### ISSUE TITLE
Error: ENOENT: no such file or directory, copyfile

### ISSUE THREAD

**Issue by timsuchanek** [yarnpkg/yarn#5275]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
When having a yarn workspace, adding one of the packages to another and then executing `yarn install` throws this error:

`Error: ENOENT: no such file or directory, copyfile '/Users/tim/Library/Caches/Yarn/v1/npm-jsonfile-4.0.0-8771aae0799b64076b76640fca058f9c10e33ecb/CHANGELOG.md' -> '/Users/tim/code/prisma-deployment-test/cli/node_modules/prisma-yml/node_modules/jsonfile/CHANGELOG.md'`

[Complete yarn-error.log](https://gist.github.com/timsuchanek/1ad127bb77b0a537efafaaf4e0d886e5)

**If the current behavior is a bug, please provide the steps to reproduce.**
```sh
git clone git@github.com:graphcool/prisma.git
git checkout 97b85c6d7ac0f8eabbb481b4e9b81f99d67d8f99
cd cli/packages/prisma-yml
yarn install
yarn build
cd ../prisma-cli-engine/
yarn add -E prisma-yml
yarn install #BOOM
```

**What is the expected behavior?**
Execute `yarn install` without throwing an error.

**Please mention your node.js, yarn and operating system version.**
`yarn@1.4.0`
`node@9.4.0`
`macOS Sierra@10.12.6`

**Comment by rally25rs** [comment]
That's a strange one. Even after it errors, I can `stat` both the "from" and "to" files reported in the error, and they both exist 😕

**Comment by nikoheikkila** [comment]
I was having a similar issue but worked around it by removing `node_modules`. Guess that's not the most friendly solution here, though.

**Comment by sth** [comment]
When I just tried to reproduce the bug, I got there error already during the "yarn add -E prisma-yml" step. The "no such file or directory" message was appropriate here since the target directory `cli/node_modules/prisma-yml/node_modules/jsonfile` really didn't exist.

**Comment by markushausammann** [comment]
`rm -rf node_modules && yarn cache clean && yarn install` solves most such problems.

**Comment by robphoenix** [comment]
I had this, and had to `rm -rf` my yarn cache dir, the one used when running `yarn` in `~/.cache/yarn` and the one in `/usr/local/share/.cache/yarn` used when running `sudo yarn`. This is in Ubuntu 18 on Windows 10.

**Comment by jkonowitch** [comment]
This is still an issue, even in `1.14`

```
Arguments: 
  /home/jeff/n/bin/node /usr/share/yarn/bin/yarn.js install

PATH: 
  /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Windows/System32:/mnt/c/Windows:/mnt/c/Windows/System32/wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0:/mnt/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/mnt/c/Windows/System32:/mnt/c/Windows:/mnt/c/Windows/System32/wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0:/mnt/c/Windows/System32/OpenSSH:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NvDLISR:/mnt/c/Program Files/Git/cmd:/mnt/c/Users/jkono/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/jkono/AppData/Local/hyper/app-2.1.2/resources/bin:/mnt/c/Users/jkono/AppData/Local/Programs/Microsoft VS Code/bin:/home/jeff/n/bin

Yarn version: 
  1.14.0

Node version: 
  10.15.1

Platform: 
  linux x64

Trace: 
  Error: ENOENT: no such file or directory, scandir '/mnt/c/Users/jkono/dev/PROJECT/node_modules/@storybook/addon-links/src'
```

**Comment by aixfox** [comment]
still errs after `rm -rf node_modules` and `yarn cache clean`.

Yarn version: 
  1.13.0

Node version: 
  11.10.0

Platform: 
  linux x64 (WSL)

Trace: 
  Error: ENOENT: no such file or directory, lstat '/home/h/.cache/yarn/v4/npm-enhanced-resolve-4.1.0-41c7e0bfdfe74ac1ffe1e57ad6a5c6c9f3742a7f/node_modules/enhanced-resolve/lib/AliasPlugin.js'

---

fix with `yarn global add yarn` to re-install yarn. (someone tells me what happens?)

**Comment by GioLogist** [comment]
I found success by reinstalling yarn w/brew.

**Comment by FabianSellmann** [comment]
Had the same problem upgrading to yarn 1.15.2. removing node modules and cleaning cache did the trick.

**Comment by kylepecci** [comment]
> still errs after `rm -rf node_modules` and `yarn cache clean`.
> 
> Yarn version:
> 1.13.0
> 
> Node version:
> 11.10.0
> 
> Platform:
> linux x64 (WSL)
> 
> Trace:
> Error: ENOENT: no such file or directory, lstat '/home/h/.cache/yarn/v4/npm-enhanced-resolve-4.1.0-41c7e0bfdfe74ac1ffe1e57ad6a5c6c9f3742a7f/node_modules/enhanced-resolve/lib/AliasPlugin.js'
> 
> fix with `yarn global add yarn` to re-install yarn. (someone tells me what happens?)

This worked for me

**Comment by alexmcmillan** [comment]
For those hitting this problem from inside WSL on Windows, this fixed it for me:

- Delete `node_modules`
- Update yarn on host Windows installation
- Run `yarn install` inside the project _from within Windows_

Installation succeeded, and switching back to an Ubuntu terminal (in WSL (inside VSCode)) I could `yarn start` without issue.

**Comment by whoughton** [comment]
This may not have anything to do with the issue here, but in case it helps, I have run across this issue when trying to copy files in between different volumes (especially NAS volumes) using the native fs routines.

**Comment by eric-burel** [comment]
I have this message from type to time when using Yalc to symlink folders

**Comment by johachi** [comment]
I got the same problem. It started occurring in my CI after refactoring a lot of files WITHOUT doing any changes to the `package.json` or `yarn.lock` file. The refactor did however change in what order the dependencies were called.

I'm unfortunately not able to share the repository since it is a private work repository. . However, I was able to solve it by having the CI run `yarn install --network-concurrency 1 --frozen-lockfile` instead of `yarn install --frozen-lockfile`.

**Comment by kira-bruneau** [comment]
It looks like this is still a problem with `--network-concurrency 1` 😕 Setting that just seems to reduce how frequently it occurs.

**Comment by simkessy** [comment]
yea ive been seeing this issue a lot lately

**Comment by brian4159** [comment]
version problem  ？

**Comment by manoharreddyporeddy** [comment]
Started with:

```
npm install --global yarn
yarn cache clean
DEL /F/Q/S node_modules > NUL 2>&1  &  RMDIR /S/Q node_modules > NUL 2>&1
```

Added below to User Environment variables
```
%LocalAppData%\Yarn\bin
```

Then
```
yarn install
```

Then since the issue was `eslint` in my case, did below
```
yarn global add eslint
yarn add eslint -D
```

Then did:
```
npm install --global eslint
npm install -D eslint
```

Then in VS Code, output, in dropdown, I saw error in `linter`:
So uninstalled below:
```
marketplace.visualstudio.com/items?itemName=fnando.linter
```
Then it worked, this was unexpected!

**Comment by akd-io** [comment]
Hitting this error consistently today.

yarn-error.log:
```
Arguments: 
  /Users/akd/.nvm/versions/node/v18.13.0/bin/node /Users/akd/.nvm/versions/node/v18.13.0/bin/yarn add @emotion/react@^11.0.0 @emotion/styled@^11.0.0 @chakra-ui/icons@^2.0.0 @chakra-ui/react@^2.0.0 @mui/material@^5.0.0 react-hook-form@^7.0.0 formik@^2.0.0 framer-motion@^9.0.0 mrm@^4.0.0 mrm-task-lint-staged@^7.0.0

PATH: 
  /Users/akd/workspace/create-next-stack/packages/create-next-stack/node_modules/.bin:/Users/akd/Library/pnpm/global/5/.pnpm/pnpm@8.3.0/node_modules/pnpm/dist/node-gyp-bin:/Users/akd/workspace/create-next-stack/node_modules/.bin:/Users/akd/Library/pnpm:/opt/homebrew/opt/openjdk/bin:/Users/akd/.nvm/versions/node/v18.13.0/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/akd/Library/pnpm:/opt/homebrew/opt/openjdk/bin:/Users/akd/.nvm/versions/node/v18.13.0/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/akd/.yarn/bin:/Users/akd/.yarn/bin:/Users/akd/.nvm/versions/node/v18.13.0/bin

Yarn version: 
  1.22.19

Node version: 
  18.13.0

Platform: 
  darwin arm64

Trace: 
  Error: ENOENT: no such file or directory, copyfile '/Users/akd/Library/Caches/Yarn/v6/npm-globby-11.1.0-bd4be98bb042f83d796f7e3811991fbe82a0d34b-integrity/node_modules/globby/index.js' -> '/Users/akd/workspace/create-next-stack-tests/run-d4eb6dcc-9846-4222-a5ef-100d98053b1e/node_modules/@typescript-eslint/typescript-estree/node_modules/globby/index.js'

npm manifest: 
  {
    "name": "run-d4eb6dcc-9846-4222-a5ef-100d98053b1e",
    "version": "0.1.0",
    "private": true,
    "scripts": {
      "dev": "next dev",
      "build": "next build",
      "start": "next start",
      "lint": "next lint"
    },
    "dependencies": {
      "@types/node": "18.15.13",
      "@types/react": "18.0.38",
      "@types/react-dom": "18.0.11",
      "eslint": "8.39.0",
      "eslint-config-next": "13.3.1",
      "next": "13.3.1",
      "react": "18.2.0",
      "react-dom": "18.2.0",
      "typescript": "5.0.4"
    }
  }

yarn manifest: 
  No manifest

Lockfile: 
  No lockfile
```

**Comment by jorgebastida** [comment]
Solved this same issue by changing the Docker volume flags from `delegated` to `cached`; It seems like there were eventual consistency issues, and files weren't where they were expected in a timely fashion.

I hope this helps somebody!

**Comment by Svashly** [comment]
My solution was removing the file mentioned on the error after the word **_copyfile_**

**Comment by perplexyves** [comment]
### TL;DR
**If all the previous solutions did not work with You, try this:**
1) Run:
```bash
yarn config set "strict-ssl" false -g
```
2) Try again the installation
